docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,550 @@
+   "source": [
+    "Finally, you can also directly use an existing MLRun function object. This is usually used when more granular control over function parameters is required (e.g. advanced parameters not supported by {py:meth}`~mlrun.projects.MlrunProject.set_function`)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7ceb2812",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "fn = mlrun.new_function(\"serving\", kind=\"serving\", image=\"mlrun/mlrun\")\n",
+    "graph = serving.set_topology(\"flow\")\n",
+    "graph.to(name=\"double\", handler=\"mylib.double\") \\\n",
+    "     .to(name=\"add3\", handler=\"mylib.add3\") \\\n",
+    "     .to(name=\"echo\", handler=\"mylib.echo\").respond()\n",
+    "\n",
+    "project.set_function(name=\"serving\", func=fn, with_repo=True)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0b2601c1",
+   "metadata": {},
+   "source": [
+    "<a id=\"customizing-functions\"></a>\n",
+    "## Customizing Functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6bed5e82",
+   "metadata": {},
+   "source": [
+    "Once you have created your MLRun function, there are many customizations you can add. Some potential customizations include:\n",
+    "\n",
+    "#### Environment Variables\n",
+    "Environment variables can be added individually, from a Python dictionary, or a file\n",
+    "\n",
+    "```python\n",
+    "# Single variable\n",
+    "fn.set_env(name=\"MY_ENV\", value=\"MY_VAL\")\n",
+    "\n",
+    "# Multiple variables\n",
+    "fn.set_envs(env_vars={\"MY_ENV\" : \"MY_VAL\", \"SECOND_ENV\" : \"SECOND_VAL\"})\n",
+    "\n",
+    "# Multiple variables from file\n",
+    "fn.set_envs(file_path=\"env.txt\")\n",
+    "```\n",
+    "\n",
+    "#### Memory, CPU, GPU Resources\n",
+    "Adding requests and limits to your function will specify what compute resources are required. It is best practice to define this for each MLRun function. See [**CPU, GPU, and memory limits for user jobs**](./configuring-job-resources.html#cpu-gpu-and-memory-limits-for-user-jobs) for more information on configuring resources.\n",
+    "\n",
+    "\n",
+    "```python\n",
+    "# Requests - lower bound\n",
+    "fn.with_requests(mem=\"1G\", cpu=1)\n",
+    "\n",
+    "# Limits - upper bound\n",
+    "fn.with_limits(mem=\"2G\", cpu=2, gpus=1)\n",
+    "```\n",
+    "\n",
+    "#### Scaling and Auto-Scaling\n",
+    "Scaling behavior can be added to real-time and distributed runtimes including `nuclio`, `serving`, `spark`, `dask`, and `mpijob`. See [**Replicas**](./configuring-job-resources.html#replicas) to see how to configure scaling behavior per runtime. This example demonstrates setting replicas for `nuclio`/`serving` runtimes:\n",
+    "\n",
+    "```python\n",
+    "# Nuclio/serving scaling\n",
+    "fn.spec.replicas = 2\n",
+    "fn.spec.min_replicas = 1\n",
+    "fn.spec.min_replicas = 4\n",
+    "```\n",
+    "\n",
+    "#### Mount Persistent Storage\n",
```

Comment:
"#### Mount persistent storage\n",
    "In some instances, you may need to mount a file-system to your container to persist data. This can be done with native K8s PVC's or the V3IO data layer for Iguazio clusters. See [**Attach storage to functions**](./function-storage.html) for more information on the storage options.\n",